silky-squirrels/chat/views.py

```diff
@@ -0,0 +1,65 @@
+from django.contrib import messages
+from django.contrib.auth.models import User
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import render, redirect
+from chat.forms import RoomCreationForm
+from chat.models import Room, RoomMember, Message
+from users.models import Profile
+
+
+@login_required
+def index(request):
+    if request.method == "POST":
+        form = RoomCreationForm(request.POST)
+        if form.is_valid():
+            room_name = form.cleaned_data["name"]
+            Room.objects.get_or_create(name=room_name)
+            return redirect("room", room_name)
+
+    else:
+        form = RoomCreationForm()
+
+    context = {
+        "form": form,
+        "friends": Profile.objects.get(user=request.user).friends.all(),
+    }
+
+    return render(request, "chat/index.html", context)
+
+
+@login_required
+def room(request, room_name, *args):
+    try:
+        room = Room.objects.get(name=room_name)
+        room_member, room_member_created = RoomMember.objects.get_or_create(user=request.user, room=room)
+        active_room_members = RoomMember.objects.filter(room=room, active=True)
```

Comment:
does all three assignment lines have the ability to raise Room.DoesNotExist error?